# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsRefreshDriver.cpp
# Commit: 85317d582166
# Full Hash: 85317d5821665149ffa385e8b55b244565c9d4c4
# Author: Tom Tung <ttung@mozilla.com>
# Date: 2020-04-07 21:44:02
# Regressor Bug: 1586761
# File Overlap Count: 1
# Description:
#   Bug 1586761 - P2 - Introduce new TimerPrecisionTypes and a set of new Reduce methods to decide the TimerPrecisionType in the nsRFPService; r=tjr
#   
#   To support checking CrossOriginIsolated in performance.now(), we need to:
#   - Add new types to TimerPrecisionType for nsRFPService
#     - System, HighResAllowed are added
# ==============================================================================

diff -r b51663641450 -r 85317d582166 layout/base/nsRefreshDriver.cpp
--- a/layout/base/nsRefreshDriver.cpp	Wed Mar 25 13:53:46 2020 +0000
+++ b/layout/base/nsRefreshDriver.cpp	Wed Mar 25 14:09:55 2020 +0000
@@ -1753,8 +1753,8 @@
           // it's currently only clamping for RFP mode. RFP mode gives a much
           // lower time precision, so we accept the security leak here for now
           if (!perf->IsSystemPrincipal()) {
-            timeStamp = nsRFPService::ReduceTimePrecisionAsMSecs(
-                timeStamp, 0, TimerPrecisionType::RFPOnly);
+            timeStamp =
+                nsRFPService::ReduceTimePrecisionAsMSecsRFP(timeStamp, 0);
           }
         }
         // else window is partially torn down already